### DISCUSSION TITLE
Expose retry count for waitForTransactionReceipt action

### DISCUSSION THREAD

**Discussion by t0rbik** [wevm/viem#1752] [Idea / Feature Request] [answered]
I would like viem to consider having `retry count` exposed for `waitForTransactionReceipt` action.

The issue is on slow blockchains and with slow rpcs recently sent transaction end up being not found in blocks with the "Transaction with {hash} could not be found" error thrown. 

The issue has been happening since the very beginning of our viem migration. And is hard to reproduce.
There were attempts to fix this in https://github.com/wevm/viem/commit/1218e977e7d9fa049f67186393b57417d7c8c6a2, several issues raised: https://github.com/wevm/viem/issues/1543, https://github.com/wevm/viem/issues/1378, https://github.com/wevm/viem/issues/1056. However, we experience it infrequently on fantom. See screenshot attached.

![image](https://github.com/wevm/viem/assets/90512605/ac4d27b2-1e6f-4c7e-a051-ff25976fca1e)

With the retry count exposed we would be able to overwrite it for some chains, such as fantom, by increasing it, to catch some more blocks after transaction submitted.

Workaround I've come up in the meantime is as follows:

```
function withMaxRetryingTime<TData, TError extends BaseError>(
  fn: () => Promise<TData>,
  {
    maxRetryingTime,
    errorInstance,
  }: {
    maxRetryingTime: number;
    errorInstance: new (..._args: any[]) => TError;
  }
) {
  return new Promise<TData>((resolve, reject) => {
    let doRetry = true;

    setTimeout(() => {
      doRetry = false;
    }, maxRetryingTime);

    const attemptRetry = async () => {
      const retry = async () => {
        attemptRetry();
      };

      try {
        const data = await fn();
        resolve(data);
      } catch (err) {
        if (doRetry && err instanceof errorInstance) return retry();
        reject(err);
      }
    };
    attemptRetry();
  });
}

const receipt = await withMaxRetryingTime(
      async () => {
        const receipt = await viemClient.waitForTransactionReceipt({
          hash: txHash,
        });
        return receipt;
      },
      {
        maxRetryingTime: 1000 * 60,
        errorInstance: TransactionNotFoundError,
      }
    );
```

**Comment by tmm** [comment]
PR welcome!

**Reply by chybisov** [reply]
Hey there! 😊 Would it be possible for you guys to take a look at adding retryCount/retryTime params to improve reliability? I believe your expertise would ensure it's done in the best way possible, enhancing the library's robustness. Thanks a bunch for considering!

**Comment by chybisov** [comment]
It is also related to https://github.com/wevm/viem/issues/1540. I would love to see `retryCount` exposed and the whole experience improved. This is really a crucial piece of functionality for many projects.

**Comment by jxom** [comment, marked as answer]
Should be in 2.7.1 once build passes.

https://viem.sh/docs/actions/public/waitForTransactionReceipt#retrycount-optional

**Reply by chybisov** [reply]
You are the GOAT! Thank you so much! 🚀

**Comment by t0rbik** [comment]
That was fast! top
